packages/common/validators/src/validators/is-birth-date-6.ts

```diff
@@ -1,5 +1,27 @@
 /** @tossdocs-ignore */
 export function isBirthDate6(birthDate: string) {
-  const re = /^[0-9]{2}(?:0[1-9]|1[0-2])(?:0[1-9]|[1,2][0-9]|3[0,1])$/;
-  return re.test(birthDate);
+  if (!/^\d{6}$/.test(birthDate)) {
+    return false;
+  }
+
+  const year = parseInt(birthDate.substring(0, 2), 10);
+  const month = parseInt(birthDate.substring(2, 4), 10);
+  const day = parseInt(birthDate.substring(4, 6), 10);
+
+  if (month < 1 || month > 12) {
+    return false;
+  }
+
+  const daysInFebruary = isLeapYear(year) ? 29 : 28;
+  const daysInMonth = [31, daysInFebruary, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
+
+  if (day < 1 || day > daysInMonth[month - 1]) {
+    return false;
+  }
+
+  return true;
+}
+
+function isLeapYear(year: number) {
+  return year !== 0 && year % 4 === 0;
 }
```

Comment:
@po4tion 👋 cc. @okinawaa 

The leap year check for the year 2000 is currently incorrect.  ex) `000229`
```ts
// Abnormal behavior 
[31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
isBirthDate6("000229"); // false

// Normal behavior 
[31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
isBirthDate6("000229"); // false
```
There are actually 29 days in February 2000.
https://ko.wikipedia.org/wiki/2000%EB%85%84_2%EC%9B%94

```suggestion
function isLeapYear(year: number) {
  return year % 4 === 0;
}
```
Wouldn't the above code be more accurate? 
Is there something I'm missing? 🤔

Reply:
@ssi02014 Thank you for finding a bug! I apply the code you suggested. fix: catch 2000y bug